perfkitbenchmarker/flag_util.py

```diff
@@ -190,3 +191,149 @@ def DEFINE_integerlist(name, default, help, on_nonincreasing=None,
+  """
+
+  def __init__(self, bytes=0):
+    self.bytes = bytes
+
+  def mult_for_base_and_exp(self, base, exp):
+    """Given base and exp, returns mult so that
+    bytes = mult * base ^ exp.
+
+    May return a float.
+    """
+
+    base_to_exp = base ** exp
+    return float(self.bytes) / base_to_exp
+
+  def exact_mult_for_base_and_exp(self, base, exp):
+    """Given base and exp, returns mult so that
+    bytes = mult * base ^ exp.
+
+    Raises ValueError if mult is not an integer.
+    """
+
+    base_to_exp = base ** exp
+    if self.bytes % base_to_exp == 0:
+      return self.bytes / base_to_exp
+    else:
+      raise ValueError('Bytes %s not divisible by %s',
+                       self.bytes, base_to_exp)
+
+  def ExactB(self):
+    return self.bytes
+
+  def ExactKiB(self):
+    return self.exact_mult_for_base_and_exp(1024, 1)
+
+  def ExactMiB(self):
+    return self.exact_mult_for_base_and_exp(1024, 2)
+
+  def ExactGiB(self):
+    return self.exact_mult_for_base_and_exp(1024, 3)
+
+  def AsB(self):
+    return self.bytes
+
+  def AsKiB(self):
+    return self.mult_for_base_and_exp(1024, 1)
+
+  def AsMiB(self):
+    return self.mult_for_base_and_exp(1024, 2)
+
+  def AsGiB(self):
+    return self.mult_for_base_and_exp(1024, 3)
+
+  def WriteWithBase(self, base):
+    """Write bytes in scientific notation.
+
+    Find mult and exp so that self.bytes = mult * base ^ exp,
+    where exp is as large as possible but <= len(SI_SUFFIXES)
+
+    Returns:
+      A tuple (mult, exp).
+    """
+
+    num = self.bytes
+    exp = 0
+
+    if num == 0:
+      return 0, 0
+
+    while num % base == 0 and exp < len(SI_SUFFIXES):
+      num = num // base
+      exp += 1
+
+    return num, exp
+
+  def __str__(self):
+    mult_ten, exp_ten = self.WriteWithBase(1000)
+    mult_two, exp_two = self.WriteWithBase(1024)
+
+    if exp_ten >= exp_two:
+      mult = mult_ten
+      exp = exp_ten
+      base = 1000
+    else:
+      mult = mult_two
+      exp = exp_two
+      base = 1024
+
+    return '%s%s%sB' % (str(mult), SI_SUFFIXES[exp],
+                        'i' if base == 1024 else '')
+
+
+  def __eq__(self, other):
+    return isinstance(other, ObjectSize) and self.bytes == other.bytes
+
+  def __lt__(self, other):
+    if isinstance(other, ObjectSize):
+      return self.bytes < other.bytes
+    else:
+      raise NotImplemented()
+
+
+OBJECT_SIZE_REGEXP = re.compile('(\d+(\.\d+)?)(K|M|G|T|P)?(i)?B')
+
+
+def ParseObjectSize(inp):
+  """Parse an object size.
+
+  The user writes a decimal followed by a suffix. Allowable base
+  suffixes are K,M,G,T, and P, meaning kilo, mega, giga, tera, and
+  peta. Each suffix may have an 'i' immediately after it, which means
+  to use the base-2 interpretation instead of base-10. All suffixes
+  must end in 'B'. All sizes must be an integer number of bytes.
+  """
+
+  match = OBJECT_SIZE_REGEXP.match(inp)
+  if match is None or match.end() != len(inp):
```

Comment:
You can force the pattern to match the entire string by putting ```$``` at the end. Similarly, you can put ```^``` at the beginning of the pattern to force the match to start at the beginning of the string.